jp/get.go

```diff
@@ -1603,6 +1603,144 @@ func (x Expr) reflectGetChild(data any, key string) (v any, has bool) {
 	return
 }
 
+func reflectGetFieldByKey(structValue reflect.Value, key string) reflect.Value {
+	if f, ok := reflectGetStructFieldByNameOrJsonTag(structValue, key); ok {
+		return structValue.FieldByIndex(f.Index)
+	}
+	return reflect.Value{}
+}
+
+func reflectGetStructFieldByNameOrJsonTag(structValue reflect.Value, key string) (result reflect.StructField, ok bool) {
+	if structValue.Type().Kind() != reflect.Struct {
+		panic("reflectGetStructFieldByNameOrJsonTag of non-struct type " + structValue.Type().String())
```

Comment:
This is a redundant check since the function is only called once the value is determined to be a struct.

Reply:
I only included the validation because parameter `structValue` is of generic type `reflect.Value`, but since the function is private, there is no problem, since it is still only called once. You are right! I removed the validation.